sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -326,7 +329,10 @@ object ScalaReflection extends ScalaReflection {
 
         val companion = t.normalize.typeSymbol.companionSymbol.typeSignature
         val cls = companion.declaration(newTermName("newBuilder")) match {
-          case NoSymbol => classOf[Seq[_]]
+          // It always returns `NoSymbol` for set classes. But we can't simply use the class of
```

Comment:
this may not be safe if a `Set` implementation doesn't have the `newBuilder` method, can we spend a little more time to investigate it? thanks!